src/Contexts/service-context.js

```diff
@@ -7,10 +7,14 @@ import { createContext, useContext, useReducer } from "react";
             return {...state , videos : (state.videos , action.payload)}
         case "SET_LIKED_VIDEOS" :
             return {...state , likedVideos : [state.likedVideos , action.payload] }
+        case  "GET_LIKED_VIDEOS" :
+            return {...state ,  likedVideos : (state.likedVideos , action.payload)}
```

Comment:
This doesn't seem like the optimised way to set State for likedVideos
I believe you are getting all the videos that are liked in the payload itself, so why are you adding "state.videos"
Also, since the payload is an array you should use the spread operator instead to update state.
Lastly, why have a separate case for "GET_LIKED_VIDEOS" when you are already setting state once in "SET_LIKED_VIDEOS" ?

Reply:
GET_LIKED_VIDEOS is used to fetch the liked videos ( get method used ) while SET_LIKED_VIDEOS adds a new video to the liked videos array.

Reply:
Used the spread operator, thank you.